.github/test_release_notes.yml

```diff
@@ -0,0 +1,8 @@
+# This file is needed for generating release notes for the Repository::testGenerateReleaseNotesWithAllArguments test
```

Comment:
Can we use the existing release notes generated here?
https://github.com/PyGithub/PyGithub/blob/main/.github/release-drafter.yml

Reply:
Note that this `release-drafter.yml` config is based on this 3rd-party GitHub action: https://github.com/release-drafter/release-drafter .

Conversely, the test in this PR is based on GitHub's own built-in release notes yaml defined here: https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes

Since the functionality introduced in this PR is specifically testing GitHub's implementation, I don't think we want to use a third-party version!

All that said, this technically probably doesn't need to go into the `.github` folder (although admittedly I haven't confirmed this).  I can move this into another folder, like something under the `tests` folder if you have thoughts for a better place!  I could also just delete this file, but keep in mind that to regenerate the test replay data, it's necessary for this file to be checked into the `main` branch before running the test (because the GitHub API itself requires this template to already be in the default branch)